src/benchmarks/micro/corefx/System.Runtime/Perf.Enum.cs

```diff
@@ -20,19 +22,534 @@ public enum Colors
+        public int CompareToEnum(Enum value, object target) => value.CompareTo(target);
+
+        [Benchmark]
+        [Arguments(BigContiguousEnum.C)]
+        public int CompareToNull(Enum value) => value.CompareTo(null);
+
+        [Benchmark]
+        [Arguments(Int32Enum.One)]
+        public TypeCode GetTypeCode(Enum value) => value.GetTypeCode();
+
+        #region IConvertible
+        [Benchmark]
+        [Arguments(ByteEnum.Max)]
+        public byte ToByte(IConvertible value) => value.ToByte(null);
+
+        [Benchmark]
+        [Arguments(SByteEnum.Max)]
+        public sbyte ToSByte(IConvertible value) => value.ToSByte(null);
+
+        [Benchmark]
+        [Arguments(Int16Enum.Max)]
+        public short ToInt16(IConvertible value) => value.ToInt16(null);
+
+        [Benchmark]
+        [Arguments(UInt16Enum.Max)]
+        public ushort ToUInt16(IConvertible value) => value.ToUInt16(null);
+
+        [Benchmark]
+        [Arguments(Int32Enum.Max)]
+        public int ToInt32(IConvertible value) => value.ToInt32(null);
+
+        [Benchmark]
+        [Arguments(UInt32Enum.Max)]
+        public uint ToUInt32(IConvertible value) => value.ToUInt32(null);
+
+        [Benchmark]
+        [Arguments(Int64Enum.Max)]
+        public long ToInt64(IConvertible value) => value.ToInt64(null);
+
+        [Benchmark]
+        [Arguments(UInt64Enum.Max)]
+        public ulong ToUInt64(IConvertible value) => value.ToUInt64(null);
+        #endregion
+
+        #region Parsing
+        [Benchmark]
+        [Arguments(typeof(Colors))]
+        public object ParseFlagsNonGeneric(Type enumType) => Enum.Parse(enumType, "Red, Orange, Yellow, Green, Blue");
+
+        [Benchmark]
+        public Colors ParseFlagsGeneric() => Enum.Parse<Colors>("Red, Orange, Yellow, Green, Blue");
+
+        [Benchmark]
+        [Arguments(typeof(ByteEnum), "Two")]
+        [Arguments(typeof(SByteEnum), "Two")]
+        [Arguments(typeof(UInt16Enum), "Two")]
+        [Arguments(typeof(Int16Enum), "Two")]
+        [Arguments(typeof(UInt32Enum), "Two")]
+        [Arguments(typeof(Int32Enum), "Two")]
+        [Arguments(typeof(UInt64Enum), "Two")]
+        [Arguments(typeof(Int64Enum), "Two")]
+        [Arguments(typeof(BigContiguousEnum), "C")]
+        [Arguments(typeof(SmallContiguousEnum), "One")]
+        public object ParseNameNonGeneric(Type enumType, string value) => Enum.Parse(enumType, value);
+
+        [Benchmark]
+        [Arguments(typeof(ByteEnum))]
+        [Arguments(typeof(SByteEnum))]
+        [Arguments(typeof(UInt16Enum))]
+        [Arguments(typeof(Int16Enum))]
+        [Arguments(typeof(UInt32Enum))]
+        [Arguments(typeof(Int32Enum))]
+        [Arguments(typeof(UInt64Enum))]
+        [Arguments(typeof(Int64Enum))]
+        public object ParseValueNonGeneric(Type enumType) => Enum.Parse(enumType, "2");
+
+        [Benchmark]
+        public ByteEnum ParseNameGenericByte() => Enum.Parse<ByteEnum>("Two");
```

Comment:
It'd be nice if Benchmark.NET supported specifying generic type arguments on methods. With that change we could collapse all of these related methods as we did with the `ParseNameNonGeneric` benchmark. I think it would require adding a `GenericTypeArguments` property on `ArgumentsAttribute`.